### ISSUE TITLE
side effects of 'each' and 'options' together

### ISSUE THREAD

**Issue by Kanaye** [astoilkov/jsblocks#126]
Look at the selects or click the "reset" button and look at the browsers console :disappointed:.
https://jsfiddle.net/h19tftc7/3/

**Comment by Kanaye** [comment]
Hmm that's an interesting problem and hard to explain (at least for me).
Let's try it:
We are calling the options-``preprocess``-function for each cycle of the ``each``-data-query.
And we are adding the ``caption`` every time as a child-element ([here](https://github.com/astoilkov/jsblocks/blob/1123d4f6ef5dfcdaba7a98765d8b1a1d5a36e30e/src/query/queries.js#L459)).
But we are only testing for ``data-role="header"`` the first time ([here](https://github.com/astoilkov/jsblocks/blob/1123d4f6ef5dfcdaba7a98765d8b1a1d5a36e30e/src/query/queries.js#L277-L315)) so the other children don't get rendered as header but as normal children.
[Here](https://jsfiddle.net/h19tftc7/4/) the example updated to show that behaviour.

We can't skip the execution of ``preprocess`` cause the ``options`` could be a property of the ``each``-data-query. Also we can't "kill" the _headers as we need them in the ``update``-function.

I don't have a clean solution (yet).

**Comment by Kanaye** [comment]
Finally fixed this 🎉.

That was harder than I imagined when I started to work on this issue again a month ago.
Had 3 *nearly* working approches, one of them totally overengineered 😆 

The acctual fix will have some negative impact on performance because we are now cloning the ``_template`` for each iteration in the ``each``-data-query to the iterations state.

I think it's not *that* big of deal bacause the VirtualElements aren't that hard to clone and we currently have time/memory-havier things to work on, but I have it on my things-to-think-over-again-list.

Also I think a a little slower but working approach should be prefered over a bugged version.

Just as some things that should be noted in my opinion.
